docs/src/main/asciidoc/hibernate-orm.adoc

```diff
@@ -168,37 +168,212 @@ Make up your mind on which approach you want to use.
+
+* Via the `packages` configuration property;
+* Via the `@io.quarkus.hibernate.orm.PersistenceUnit` package-level annotation.
+
+If both are mixed, the annotations are ignored and only the `packages` configuration properties are taken into account.
+
+Using the `packages` configuration property is simple:
+
+[source,properties]
+----
+quarkus.hibernate-orm.database.generation=drop-and-create
+quarkus.hibernate-orm.packages=org.acme.model.defaultpu
+
+quarkus.hibernate-orm."users".database.generation=drop-and-create
+quarkus.hibernate-orm."users".datasource=users
+quarkus.hibernate-orm."users".packages=org.acme.model.user
+----
+
+This configuration snippet will create two persistence units:
+
+* The default one which will contain all the model classes under the `org.acme.model.defaultpu` package, subpackages included.
+* A named `users` persistence unit which will contain all the model classes under the `org.acme.model.user` package, subpackages included.
+
+It is possible to attach several packages to a persistence unit:
+
+[source,properties]
+----
+quarkus.hibernate-orm."users".packages=org.acme.model.shared,org.acme.model.user
+----
+
+All the model classes under the `org.acme.model.shared` and `org.acme.model.user` packages will be attached to the `users` persistence unit.
+
+It is also supported to attach a given model class to several persistence units.
+
+[NOTE]
+====
+Model classes need to be consistently added to a given persistence unit.
+Meaning all dependent model classes of a given entity (mapped super classes, embeddables...) are required to be attached to the persistence unit.
+As we are dealing with the persistence unit at the package level, it should be simple enough.
+====
+
+[WARNING]
+====
+Panache entities can be attached to only one persistence unit.
+
+For entities attached to several persistence units, you cannot use Panache.
+You can mix the two approaches though and mix Panache entities and traditional entities where multiple persistence units are required.
+
+If you have a use case for that and clever ideas about how to implement it without cluttering the simplified Panache approach,
+contact us on the link:{quarkus-mailing-list-index}[quarkus-dev mailing list].
+====
+
+The second approach to attach model classes to a persistence unit is to use package-level `@io.quarkus.hibernate.orm.PersistenceUnit` annotations.
+Again, the two approaches cannot be mixed.
+
+To obtain a configuration similar to the one above with the `packages` configuration property, create a `package-info.java` file with the following content:
+
+[source,java]
+----
+@PersistenceUnit("users") <1>
+package org.acme.model.user;
+
+import io.quarkus.hibernate.orm.PersistenceUnit;
+----
+<1> Be careful, use the `@io.quarkus.hibernate.orm.PersistenceUnit` annotation, not the JPA one.
+
+[CAUTION]
+====
+We only support setting `@PersistenceUnit` at the package level,
```

Comment:
Package, field and parameter levels according to the sources :wink:

Reply:
No. It's used for field and parameters for injection. Not to attach entities to the persistence units. Thus the precision and the second part of the sentence.